dbms/src/Functions/FunctionsRound.h

```diff
@@ -1186,7 +1201,7 @@ struct TiDBDecimalRound
             scaled_value *= PowForOutput::result[-difference];
 
         // check overflow and construct result.
-        if (scaled_value > DecimalMaxValue::get(info.output_prec))
+        if (scaled_value > DecimalMaxValue::get(info.output_prec + carry))
```

Comment:
IMO it's safe to always take carry into accout: `if (scaled_value > DecimalMaxValue::get(info.output_prec + 1))`.

Reply:
I aggree, just make sure `info.output_prec + 1 <= max_decimal_prec`